src/Extensions/Hosting.Services/Internal/OmexServiceRunner.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Fabric;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services
+{
+	internal abstract class OmexServiceRunner<TService, TContext> : IOmexServiceRunner
+		where TService : IServiceFabricService<TContext>
+		where TContext : ServiceContext
+	{
+		protected readonly string ApplicationName;
+		protected readonly ServiceContextAccessor<TContext> ContextAccessor;
+		public IEnumerable<IListenerBuilder<TService>> ListenerBuilders { get; }
+		public IEnumerable<IServiceAction<TService>> ServiceActions { get; }
+		public OmexServiceRunner(
+			IHostEnvironment environment,
+			ServiceContextAccessor<TContext> contextAccessor,
+			IEnumerable<IListenerBuilder<TService>> listenerBuilders,
+			IEnumerable<IServiceAction<TService>> serviceActions)
+		{
+			ApplicationName = environment.ApplicationName;
```

Comment:
>ApplicationName [](start = 3, length = 15)

I can see input validation is skipped in many places. Are these objects the new nullable classes?

Reply:
In most of the cases they are coming from the DI and it won't pass nulls, in other cases, it's about non-nullable classes, and in some cases, it's skipped because it's not supposed to be used publicly

Reply:
I think it is a good practice to validate these. DI frameworks differ and some of them could pass null. I think the validation can save some time troubleshooting in the future.

---
In reply to: [399421207](https://github.com/microsoft/Omex/pull/180#discussion_r399421207) [](ancestors = 399421207)

Reply:
I think it's enforced on ServiceProvider level so any DI that implements it should throw if it's unable to resolve the dependency. And anyway we should not use any DI that could path null. But I agree that we should consider the general way of validating inputs that are coming from users of the library.